src/main/java/blusunrize/immersiveengineering/common/data/RecipeSerializers.java

```diff
@@ -0,0 +1,41 @@
+package blusunrize.immersiveengineering.common.data;
+
+import blusunrize.immersiveengineering.ImmersiveEngineering;
+import blusunrize.immersiveengineering.common.crafting.FlareBulletRecipe;
+import blusunrize.immersiveengineering.common.crafting.PotionBulletRecipe;
+import blusunrize.immersiveengineering.common.crafting.SpeedloaderRecipe;
+import net.minecraft.item.crafting.IRecipeSerializer;
+import net.minecraft.item.crafting.SpecialRecipeSerializer;
+import net.minecraftforge.event.RegistryEvent.Register;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+
+@Mod.EventBusSubscriber(modid = ImmersiveEngineering.MODID, bus = Mod.EventBusSubscriber.Bus.MOD)
+public class RecipeSerializers
+{
+	public static final SpecialRecipeSerializer<SpeedloaderRecipe> SPEEDLOADER_LOAD = IRecipeSerializer.register(ImmersiveEngineering.MODID+":crafting_special_speedloader_load", new SpecialRecipeSerializer<>(SpeedloaderRecipe::new));
+	public static final SpecialRecipeSerializer<PotionBulletRecipe> POTION_BULLET_FILL = IRecipeSerializer.register(ImmersiveEngineering.MODID+":crafting_special_potion_bullet_fill", new SpecialRecipeSerializer<>(PotionBulletRecipe::new));
+	public static final SpecialRecipeSerializer<FlareBulletRecipe> FLARE_BULLET_COLOR = IRecipeSerializer.register(ImmersiveEngineering.MODID+":crafting_special_flare_bullet_color", new SpecialRecipeSerializer<>(FlareBulletRecipe::new));
+
+	@SubscribeEvent
+	public static void register(Register<IRecipeSerializer<?>> event)
+	{
+		Field[] fields = RecipeSerializers.class.getDeclaredFields();
```

Comment:
Please change this to not use reflection. Forge has `DeferredRegister` (see Lex's mods for examples), that should make this doable without much code duplication.

Reply:
done, but please double check that I did that correctly.
I mean everything appears to be working, but still